src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.IncrementalMemberEditAnalyzer.cs

```diff
@@ -219,6 +219,14 @@ public void UpdateDocumentWithCachedDiagnostics(Document document)
                 var members = syntaxFacts.GetMethodLevelMembers(root);
                 var memberSpans = members.SelectAsArray(member => member.FullSpan);
                 var changedMemberId = members.IndexOf(changedMember);
+
+                // The changed member might not be a method level member (e.g. a class).
+                // We can't perform method analysis  on these so we bail out.
+                if (changedMemberId == -1)
```

Comment:
Above we look at all method level members and index into that.  However I sometimes got back a class declaration in my testing, which gave a -1 as the index and this passed it along until it used the index later on.

This check seems necessary to me, but want to confirm with @mavasani that I could get the class node as the changed member.

Reply:
any way to test?

Reply:
We need to probably write some test accessors to be able to write unit tests. Let me open a tracking test issue.

Reply:
https://github.com/dotnet/roslyn/issues/65678